### PR TITLE
Fix/improve table sorting

### LINKED ISSUE #1370: OcTableFiles: Sorting is no working by clicking on "Share owner" column header
### Steps to reproduce
1. Click on the header "Share owner"

### Expected behaviour
The table is sorted in alphabetical ascending or (by the second click) descending order by share owner's surname as the the default user icons also show the first letter of the surname

### Actual behaviour
Clicking has no effect

### PR DESCRIPTION
**Pull request by pascalwengerter**
## Description
Currently, same sized arrays containing objects don't get properly sorted. We've added the capability to sort them based on their keys by passing a string or function

## Related Issue
- Fixes #1370

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Checklist:
- [X] Code changes

## Open tasks:
- [X] Unit tests added

### CODE REVIEW

File: src/components/table/mixins/sort.js [hunk 1]
```diff
@@ -21,10 +21,26 @@ export default {
...
+
+        if (sortable) {
+          if (typeof sortable === "string") {
+            const genArrComp = vals => {
+              return vals.map(val => val[sortable]).join("")
+            }
+
+            aValue = genArrComp(aValue)
+            bValue = genArrComp(bValue)
+          } else if (typeof sortable === "function") {
```
Comment (fschade) [inline src/components/table/mixins/sort.js:38]: can we add one test for function sortable?

Comment (pascalwengerter) [reply src/components/table/mixins/sort.js:38]: Agreed that we should, do you have any idea of a use case?

**Review by kulmann — APPROVED** [review]
👍

File: src/components/table/mixins/sort.js (hunk 1 above)
Comment (fschade) [reply src/components/table/mixins/sort.js:38]: Maybe a star ⭐️ rating ?

Comment (pascalwengerter) [reply src/components/table/mixins/sort.js:38]: The branch is merged but I raised an issue [here](https://github.com/owncloud/owncloud-design-system/issues/1437) if you want to get funky :)

**Pull request merged**